GitUpKit/Extensions/GCRepository+Index.m

```diff
@@ -39,22 +39,61 @@ - (BOOL)resetIndexToHEAD:(NSError**)error {
 }
 
 - (BOOL)removeFileFromIndex:(NSString*)path error:(NSError**)error {
+  return [self removeFilesFromIndex:@[ path ] error:error];
+}
+
+- (BOOL)removeFilesFromIndex:(NSArray<NSString*>*)paths error:(NSError**)error {
   GCIndex* index = [self readRepositoryIndex:error];
   if (index == nil) {
     return NO;
   }
-  return [self removeFile:path fromIndex:index error:error] && [self writeRepositoryIndex:index error:error];
+
+  for (NSString* path in paths) {
+    if (![self removeFile:path fromIndex:index error:error] || (error && *error != nil)) {
+      [self writeRepositoryIndex:index error:error];
+      return false;
+    }
+  }
+
+  return [self writeRepositoryIndex:index error:error];
 }
 
 - (BOOL)addFileToIndex:(NSString*)path error:(NSError**)error {
+  return [self addFilesToIndex:@[ path ] error:error];
+}
+
+- (BOOL)addFilesToIndex:(NSArray<NSString*>*)paths error:(NSError**)error {
   GCIndex* index = [self readRepositoryIndex:error];
   if (index == nil) {
     return NO;
   }
-  return [self addFileInWorkingDirectory:path toIndex:index error:error] && [self writeRepositoryIndex:index error:error];
+
+  BOOL failed = NO;
+  BOOL shouldWriteRepository = NO;
+  for (NSString* path in paths) {
+    if (![self addFileInWorkingDirectory:path toIndex:index error:error] || (error && *error != nil)) {
+      failed = YES;
+      break;
+    }
+
+    shouldWriteRepository = true;
+  }
+
+  if (failed && shouldWriteRepository) {
+    if (shouldWriteRepository) {
+      [self writeRepositoryIndex:index error:NULL];
+    }
+    return FALSE;
```

Comment:
`return NO`

Reply:
ugh I started this branch 3 years ago and it shows 😂 I should've caught these when reviewing though